scripts/pages/map-selector/map-entry.ts

```diff
@@ -178,4 +183,20 @@ class MapEntryHandler {
 			cp.SetHasClass('map-entry--new', false);
 		}
 	}
+
+	updatePlayerCount() {
+		// Roaming lobby player count
+		const playerCount = this.panels.cp.mapData.roamingLobbyPlayerCount;
+		if (playerCount > 0) {
+			this.panels.lobbyContainer.visible = true;
+			this.panels.cp.SetDialogVariableInt('playerCount', playerCount);
+		} else {
+			this.panels.lobbyContainer.visible = false;
+		}
+	}
+
+	metaModeChanged() {
+		// Wait for next roaming lobby update
+		this.panels.lobbyContainer.visible = false;
+	}
```

Comment:
C++ should post `MapCache_RoamingLobbiesUpdated` when clearing the lobby counts, then pano only has to be concerned with listening for that event and the higher-level logic is left to the map selector code (and isn't duplicated). This is how pano stuff should generally go (C++ provides some data for pano and fires an event when that data changes so pano knows to update itself)